presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -380,12 +380,18 @@ private static void assertJsonNodesHaveStats(JsonRenderer.JsonRenderedNode node)
     public void testExplainAnalyzeFormatJson()
     {
         JsonRenderer renderer = new JsonRenderer(getQueryRunner().getMetadata().getFunctionAndTypeManager());
-        Map<PlanFragmentId, JsonRenderer.JsonPlan> fragments = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT * FROM orders").getOnlyValue());
-        fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
-        fragments = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT rank() OVER (PARTITION BY orderkey ORDER BY clerk DESC) FROM orders").getOnlyValue());
-        fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
-        fragments = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT rank() OVER (PARTITION BY orderkey ORDER BY clerk DESC) FROM orders WHERE orderkey < 0").getOnlyValue());
-        fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
+        List<Map<PlanFragmentId, JsonRenderer.JsonPlan>> fragmentsList = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT * FROM orders").getOnlyValue());
+        for (Map<PlanFragmentId, JsonRenderer.JsonPlan> fragments : fragmentsList) {
+            fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
+        }
+        fragmentsList = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT rank() OVER (PARTITION BY orderkey ORDER BY clerk DESC) FROM orders").getOnlyValue());
+        for (Map<PlanFragmentId, JsonRenderer.JsonPlan> fragments : fragmentsList) {
+            fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
+        }
+        fragmentsList = renderer.deserialize((String) computeActual("EXPLAIN ANALYZE (format JSON) SELECT rank() OVER (PARTITION BY orderkey ORDER BY clerk DESC) FROM orders WHERE orderkey < 0").getOnlyValue());
+        for (Map<PlanFragmentId, JsonRenderer.JsonPlan> fragments : fragmentsList) {
+            fragments.values().forEach(planFragment -> assertJsonNodesHaveStats(planFragment.getPlan()));
+        }
```

Comment:
we should add another test where we enable CTEs and verify that all stages exist in the plan output

Reply:
added